integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ETagSupportTestCase.java

```diff
@@ -0,0 +1,141 @@
+
+        // Setting initial http headers
+        headers = new HashMap<String, String>();
+        headers.put("action", "urn:echoString");
+        headers.put("Content-Type", "application/soap+xml");
+        headers.put("charset", "UTF-8");
+
+        payload = "<soapenv:Envelope xmlns:soapenv=\"http://www.w3.org/2003/05/soap-envelope\">" +
+                  "<soapenv:Body>" +
+                  "<p:echoInt xmlns:p=\"http://echo.services.core.carbon.wso2.org\">" +
+                  "<in>1</in>" +
+                  "</p:echoInt>" +
+                  "</soapenv:Body>" +
+                  "</soapenv:Envelope>";
+
+    }
+
+    @Test(groups = "wso2.esb", description = "ETag enable test for a pass-through proxy", enabled = true)
+    public void testEnableETag() throws Exception {
+
+        HttpResponse response = httpClient.doPost(epr, headers, payload, "text/xml");
+
+        eTagHeader = response.getFirstHeader("ETag");
+        Assert.assertNotNull(eTagHeader, "ETag header is not present in the response");
+
+        int statusCode = response.getStatusLine().getStatusCode();
+        Assert.assertEquals(statusCode, 200, "Incorrect status code. Expected: 200 Found: " + statusCode);
+        Assert.assertNotNull(httpClient.getResponsePayload(response), "Payload should not empty");
+    }
+
+    @Test(groups = "wso2.esb", description = "If-None-Match header test for cache hit", enabled = false)
+    public void testIfNoneMatchHeaderCacheHit() throws Exception {
+        headers.put("If-None-Match", "\"" + eTagHeader.getValue() + "\"");
+
+        HttpResponse response = httpClient.doPost(epr, headers, payload, "text/xml");
+
+        int statusCode = response.getStatusLine().getStatusCode();
+        Assert.assertEquals(statusCode, 304, "Incorrect status code. Expected: 304 Found: " + statusCode);
+        Assert.assertNotNull(httpClient.getResponsePayload(response), "Payload should not empty");
+    }
+
+    @Test(groups = "wso2.esb", description = "If-None-Match header test for cache miss", enabled = false)
+    public void testIfNoneNoneMatchHeaderCacheMiss() throws Exception {
+
+        payload = "<soapenv:Envelope xmlns:soapenv=\"http://www.w3.org/2003/05/soap-envelope\">" +
+                  "<soapenv:Body>" +
+                  "<p:echoInt xmlns:p=\"http://echo.services.core.carbon.wso2.org\">" +
+                  "<in>2</in>" +
+                  "</p:echoInt>" +
+                  "</soapenv:Body>" +
+                  "</soapenv:Envelope>";
+
+        HttpResponse response = httpClient.doPost(epr, headers, payload, "text/xml");
+
+        int statusCode = response.getStatusLine().getStatusCode();
+        Assert.assertEquals(statusCode, 200, "Incorrect status code. Expected: 200 Found: " + statusCode);
+        Assert.assertNotNull(httpClient.getResponsePayload(response), "Payload should not empty");
+
+        Assert.assertTrue(!response.getFirstHeader("ETag").getValue().equals(headers.get("If-None-Match")),
```

Comment:
We can use assertNotEquals for this. Also let's print the whole response in the case of a failure, that's make debugging easy in a failure